go-controller/pkg/ovn/base_secondary_layer2_network_controller.go

```diff
@@ -96,7 +98,17 @@ func (h *baseSecondaryLayer2NetworkControllerEventHandler) AddResource(obj inter
 		if !ok {
 			return fmt.Errorf("could not cast %T object to Node", obj)
 		}
-		return h.oc.addUpdateNodeEvent(node)
+		if h.oc.isLocalZoneNode(node) {
+			var nodeParams *nodeSyncs
+			if fromRetryLoop {
+				_, syncMgmtPort := h.oc.mgmtPortFailed.Load(node.Name)
+				nodeParams = &nodeSyncs{syncMgmtPort: syncMgmtPort}
+			} else {
+				nodeParams = &nodeSyncs{syncMgmtPort: true}
+			}
+			return h.oc.addUpdateLocalNodeEvent(node, nodeParams)
+		}
+		return h.oc.addUpdateRemoteNodeEvent(node)
```

Comment:
the baseSecondaryLayer2 runs for localnet too, we should skip if we are at localnet.

Reply:
Well, addUpdateLocalNodeEvent() checks for:

```
if util.IsNetworkSegmentationSupportEnabled() && oc.IsPrimaryNetwork() {
```

Isn't that enough?

Reply:
@dceara I have create a draft PR to illustrate a way to allow to override event handler at final controllers:
- https://github.com/ovn-org/ovn-kubernetes/pull/4501

Reply:
After trying it out, it's not that straightforward, we'd have to do some ugly tricks to access L2/localnet controller functions from inside the base handler code.  Decided to go with the duplication of handlers for L2 and Localnet.